### PR TITLE
Implement java precommit dataflow examples tests to run on java 11

### PR DESCRIPTION
**Pull request by pawelpasterz**
Implement PreCommit Dataflow Examples to run on java 11

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by mwalenia** [comment]
run seed job

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by mwalenia** [comment]
run seed job

**Comment by mwalenia** [comment]
Run java precommit

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by mwalenia** [comment]
LGTM, when the tests finish, I'll merge it. Thanks for your contribution!

**Comment by mwalenia** [comment]
Run Java PreCommit

**Comment by mwalenia** [comment]
Run Java PreCommit

**Comment by mwalenia** [comment]
Run Java PreCommit

**Comment by mwalenia** [comment]
Run Java PreCommit

**Comment by mwalenia** [comment]
run java precommit

**Comment by mwalenia** [comment]
retest this please

**Comment by mwalenia** [comment]
retest this please

**Comment by kkucharc** [comment]
run seed job

**Comment by kkucharc** [comment]
retest this please

**Comment by kkucharc** [comment]
Run Java PreCommit

**Comment by mwalenia** [comment]
@kennknowles do you have anything to add to the review, or can we merge this?

**Comment by aaltay** [comment]
@kennknowles or @Ardagan - could either one of you review this PR please?

**Comment by kennknowles** [comment]
Sorry - got this mixed up with a similar PR and lost track.

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        ],
+        timeoutMins: 30,
+)
+builder.build {
+    publishers {
+        archiveJunit('**/build/test-results/**/*.xml')
+    }
+
+    steps {
+        gradle {
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy]: Could this step be a publish?

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            tasks 'javaExamplesDataflowPrecommit'
+            switches "-Dorg.gradle.java.home=${commonJobProperties.JAVA_8_HOME}"
+            switches '-PdisableSpotlessCheck=true'
+            switches '-x runLegacyWorkerPreCommitTest'
+            switches '-x runApplianceTest'
+            switches '-x runWindmillTest'
+            commonJobProperties.setGradleSwitches(delegate)
+        }
+
+        gradle {
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy]: And then this step point to the deps from the prior step?

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            rootBuildScriptDir(commonJobProperties.checkoutDir)
+            tasks ':runners:google-cloud-dataflow-java:examples:runLegacyWorkerPreCommitTest'
+            tasks ':runners:google-cloud-dataflow-java:examples-streaming:runApplianceTest'
+            tasks ':runners:google-cloud-dataflow-java:examples-streaming:runWindmillTest'
+            switches '-x shadowJar'
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy]: Doing these exclusions seems a fragile. Is there a way to indicate that some dependencies are provided from published cache and not built?

Comment (pawelpasterz) [reply .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy]: @kennknowles thanks for all comments, first I thought I will answer to all your comments but then I decided to gather it all in just one. I can see there is lot of knowledge that is missing to me right now, can we contact outside this pr scope? (ex email?) I would very appreciate if you will share some knowledge with me. What do you think?

**Comment by mwalenia** [comment]
retest this please

**Comment by mwalenia** [comment]
run seed job

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by mwalenia** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by mwalenia** [comment]
Run Java PreCommit

**Pull request closed without merging**